### PR TITLE
Add file configuration examples

### LINKED ISSUE #3465: Add end to end usage example for file configuration
Would an example in this section be appropriate? It might help make it more clear how a Configurer would be instantiated and used

_Originally posted by @codeboten in https://github.com/open-telemetry/opentelemetry-specification/pull/3437#discussion_r1179452191_

### PR DESCRIPTION
**Pull request by jack-berg**
Resolves #3465

### CODE REVIEW

File: specification/configuration/file-configuration.md [hunk 1]
```diff
@@ -350,6 +350,90 @@ register [component providers](#component-provider).
...
+```
+
+### Via OTEL_EXPERIMENTAL_CONFIG_FILE
+
+If an SDK
+supports [OTEL_EXPERIMENTAL_CONFIG_FILE](./sdk-environment-variables.md#file-configuration),
+then setting `OTEL_EXPERIMENTAL_CONFIG_FILE` provides a simple way to obtain an
+SDK initialized from the specified config file. The pattern for accessing the
+configured SDK components and installing into instrumentation will vary by
+language. For example, the usage in java might resemble:
```
Comment (reyang) [inline specification/configuration/file-configuration.md:416]:
```suggestion
language. For example, the usage in Java might resemble:
```

File: specification/configuration/file-configuration.md [hunk 2]
```diff
@@ -350,6 +350,90 @@ register [component providers](#component-provider).
...
+try {
+    // Parse local and remote configuration files to configuration models
+    OpenTelemetryConfiguration localConfigurationModel = FileConfiguration.parse(new File("/app/sdk-config.yaml"));
+    OpenTelemetryConfiguration remoteConfigurationModel = FileConfiguration.parse(getRemoteConfiguration("http://example-host/config/my-application"));
+    
+    // Merge the configuration models using custom logic
+    OpenTelemetryConfiguration resolvedConfigurationModel = merge(localConfigurationModel, remoteConfigurationModel);
+    
+    // Create SDK components from resolved configuration model
+    openTelemetry = FileConfiguration.create(configurationModel);
```
Comment (reyang) [inline specification/configuration/file-configuration.md:397]:
```suggestion
    openTelemetry = FileConfiguration.create(resolvedConfigurationModel);
```

**Review by reyang — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

File: specification/configuration/file-configuration.md [hunk 3]
```diff
@@ -350,6 +350,90 @@ register [component providers](#component-provider).
...
+variety of ways to achieve simple or complex configuration goals.
+
+For example, a simple case would consist of calling `Parse` with a configuration
+file, and passing the result to `Create` to obtain configured SDK components:
+
+```java
+OpenTelemetry openTelemetry = OpenTelemetry.noop();
+try {
+    // Parse configuration file to configuration model
+    OpenTelemetryConfiguration configurationModel = FileConfiguration.parse(new File("/app/sdk-config.yaml"));
```
Comment (pyohannes) [inline specification/configuration/file-configuration.md:368]: Is `OpenTelemetryConfiguration` a type specific to file-based configuration? Or would this also be used to represent in-memory configuration?

Comment (jack-berg) [reply specification/configuration/file-configuration.md:368]: This is the meant to be the java representation of the in-memory configuration model.

Comment (jack-berg) [reply specification/configuration/file-configuration.md:368]: >It's not clear to me if I'd also need to use FileConfiguration.create when building the whole configuration in-memory

The contract of `Create` is that it takes an in memory configuration model and returns configured SDK components. The easiest way to obtain an in memory configuration model is to use `Parse`, but it should be possible to build one programmatically as well. `Create` should not care about how the in memory configuration model was created.

**Review by trask — APPROVED** [review]

**Pull request merged**